aicsimageio/readers/tiff_reader.py

```diff
@@ -42,6 +75,156 @@ def _is_this_type(buffer: io.BufferedIOBase) -> bool:
+                sample = scenes[0].pages[0].asarray()
+
+                # Combine length of scenes and operating shape
+                # Replace YX dims with empty dimensions
+                operating_shape = (len(scenes), *operating_shape)
+                operating_shape = operating_shape[:-2] + (1, 1)
+
+                # Make ndarray for lazy arrays to fill
+                lazy_arrays = np.ndarray(operating_shape, dtype=object)
+                for all_page_index, np_tuple in enumerate(np.ndenumerate(lazy_arrays)):
+                    # Unpack np_tuple
+                    np_index, _ = np_tuple
+
+                    # Scene index is the first index in np_index
+                    scene_index = np_index[0]
+
+                    # This page index is current enumeration divided by scene index + 1
+                    # For example if the image has 10 Z slices and 5 scenes, there would be 50 total pages
+                    this_page_index = all_page_index // (scene_index + 1)
+
+                    # Fill the numpy array with the delayed arrays
+                    lazy_arrays[np_index] = da.from_delayed(
+                        delayed(TiffReader._imread)(self._file, scene_index, this_page_index),
+                        shape=sample.shape,
+                        dtype=sample.dtype
+                    )
+
+                # Convert the numpy array of lazy readers into a dask array
+                data = da.block(lazy_arrays.tolist())
+
+                # Only return the scene dimension if multiple scenes are present
+                if len(scenes) == 1:
+                    data = data[0, :]
+
+                # Set _dask_data
+                self._dask_data = data
+
+        return self._dask_data
+
+    def load_slice(self, slice_index: int = 0) -> np.ndarray:
+        with TiffFile(self._file) as tiff:
+            return tiff.asarray(key=slice_index)
+
+    def dtype(self):
+        if self._dtype is None:
+            with TiffFile(self._file) as tiff:
+                self._dtype = tiff.pages[0].dtype
+
+        return self._dtype
+
+    @property
+    def dims(self) -> str:
+        if self._dims is None:
+            # Get a single scenes dimensions in order
+            with TiffFile(self._file) as tiff:
+                single_scene_dims = tiff.series[0].pages.axes
+
+                # We can sometimes trust the dimension info in the image
+                if all([d in Dimensions.DefaultOrder for d in single_scene_dims]):
+                    # Add scene dimension only if there are multiple scenes
+                    if len(tiff.series) == 1:
+                        self._dims = single_scene_dims
+                    else:
+                        self._dims = f"{Dimensions.Scene}{single_scene_dims}"
+                # Sometimes the dimension info is wrong in certain dimensions, so guess that dimension
+                else:
+                    guess = self.guess_dim_order(tiff.series[0].pages.shape)
```

Comment:
Should there be a log message here for that? (or is that within `guess_dim_order`?)